Core/GDCore/Events/Builtin/ForEachEvent.cpp

```diff
@@ -37,6 +37,18 @@ vector<gd::Expression*> ForEachEvent::GetAllExpressions() {
 
   return allExpressions;
 }
+
+vector<pair<gd::Expression*, gd::ParameterMetadata> >
+    ForEachEvent::GetAllExpressionsWithMetadata() {
+  vector<pair<gd::Expression*, gd::ParameterMetadata> >
+      allExpressionsWithMetadata;
+  auto type = gd::ParameterMetadata().SetType("object");

```

Comment:
nitpicking: I would name this `metadata` or `parameterMetadata` rather than type